artiq/master/worker_db.py

```diff
@@ -90,6 +91,11 @@ def get(self, name):
             raise DeviceError("Failed to create device '{}'"
                               .format(name)) from e
         self.active_devices.append((desc, dev))
+
+        if "channel" in desc["arguments"]:
+            key = "channel " + str(desc["arguments"]["channel"])
+            self.lib[key] = key + " (" + name + ")"
```

Comment:
Keep that function simple and move this code out of it.
``lib`` is not a good attribute name and I don't see why we should have it at all anyway. Just scan the DDB when the exception occurs. This is not performance-critical code.

Reply:
It's also very poor architecture to have "channel X" strings in that dictionary...